tests/network/primary_pod_network.go

```diff
@@ -107,7 +107,10 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 				It("should report PodIP/s as its own on interface status", func() {
 					vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, vmi.Namespace)
 					Eventually(vmiIP).Should(WithTransform(removeCIDR, Equal(vmiPod.Status.PodIP)), "should contain VMI Status IP as Pod status ip")
-					Eventually(vmiIPs).Should(ContainElements(vmi.Status.Interfaces[0].IP, libnet.DefaultIPv6Address), "should contain at IPs the IP and the cloud-init configured ipv6")
+
+					// Check first fixed IPv6 address so we are sure that GA has quick in
+					Eventually(vmiIPs).Should(ContainElement(libnet.DefaultIPv6Address), "should contain IPv6 address set by cloud-init and reported by guest agent")
+					Expect(vmi.Status.Interfaces[0].IPs).Should(ContainElement(vmi.Status.Interfaces[0].IP), "should contain IPv4 reported by guest agent")
```

Comment:
It might be preferable to also validate `vmi.Status.Interfaces[0].IP` is a legit IP (or IP/mask until the mask is removed),
not only listed at `mi.Status.Interfaces[0].IPs`.

Wdyt ?

Reply:
Is not needed to fix the race condition, and this is the only purpose of this PR, I am not going to check the IP content here.